### PR TITLE
PHOENIX-6692 Add HBase 2.5 support

### PR DESCRIPTION
**Pull request by stoty**

### CODE REVIEW

**Comment by stoty** [comment]
We have test failures, need too look into them.
[ERROR] Errors:
[ERROR]   MaxLookbackIT.testRecentMaxVersionsNotCompactedAway:315->majorCompact:343 » TestTimedOut
[ERROR]   MaxLookbackIT.testRecentlyDeletedRowsNotCompactedAway:154->majorCompact:343 » TestTimedOut
[ERROR]   MaxLookbackIT.testTTLAndMaxLookbackAge:238->majorCompact:343 » TestTimedOut te...
[ERROR]   TransformMonitorExtendedIT.testTransformIndexWithNamespaceEnabled:123 » SQL ER...
[ERROR]   TransformMonitorExtendedIT.testTransformTableWithNamespaceEnabled:155 » SQL ER...
[ERROR]   TransformMonitorExtendedIT.testTransformWithGlobalAndTenantViews:246 » SQL ERR...

**Comment by stoty** [comment]
This is not expected to compile on Jenkins until official 2.5.0 artifacts are available.
While we recompile 2.4 with Hadoop3 to run the tests, the other compatibility modules use the public artifacts for compilation, and will fail.

For compiling the compatibility modules Hadoop2/Hadoop3 is not an issue, the Hadoop3 compiled HBase is only needed for the shaded JARs and the minicluster tests.

**Review by richardantal — COMMENTED** [review]
Looks good to me
I built HBase 2.5.0 locally and than ran `mvn clean package -DskipTests -Dhbase.profile=2.5` successfully.

Locally I had some test failures but It fails with the default 2.4 HBase profiles as well, probably something is not OK on my side.

File: phoenix-hbase-compat-2.5.0/pom.xml [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    </dependency>
+    <dependency>
+      <groupId>org.apache.hbase</groupId>
+      <artifactId>hbase-metrics-api</artifactId>
+      <version>${hbase25.compat.version}</version>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-api</artifactId>
```
[the remark this answers is not in the document]
Comment (stoty) [reply phoenix-hbase-compat-2.5.0/pom.xml:102]: That version is defined in the main pom.

Also, even though 2.5.0 hasn't been released yet, branch-2.5 identified itself as 2.5.1. I expect that it's going to be reset
before release, but for now you need to override the version while building:

`mvn clean package -Dhbase.profile=2.5 -Dhbase.version=2.5.1-SNAPSHOT`

[the remark this answers is not in the document]
Comment (richardantal) [reply phoenix-hbase-compat-2.5.0/pom.xml:102]: I used the 2.5.0RC0 branch/tag which had 2.5.0 version.

**Comment by stoty** [comment]
2.5.0 is out.
I have updated the CI setup to also build for 2.5, and also added 5.2.0 to the upgrade tests for Hbase 2.3 and 2.4.

I think we can commit this now. (Once the tests have run)

**Comment by stoty** [comment]
PTAL @gjacoby126 @richardantal

**Comment by stoty** [comment]
Fixed some checkstlye issues.

**Review by gjacoby126 — APPROVED** [review]
+1, thanks @stoty

**Pull request closed without merging**